src/on.js

```
@@ -0,0 +1,171 @@
+            cb(data.toObject())
+        }
+    }
+
+    return {
+        onError,
+        onProcessingError,
+        onMessage
+    }
+}
+
+/**
+ * Assembles the dependencies into the `On` function that is used for listening to events from PubSub
+ * 
+ * @param {Object} config The injected dependencies
+ * 
+ * @returns {Object} A reference to On and onMessage. 
+ */
+function createOn({Consumer, region, account, service, deployment, subscriptions = new Set()}) {
+    const {
+        deserialize,
+        buildQueueUrl
+    } = createActions()
+
+    const {
+        onError,
+        onProcessingError,
+        onMessage,
+    } = createHandlers({deserialize})
+
+    /**
+     * On will take an event name and setup a listener for that event. It will automagically figure out the event type
+     * and deserialize it into a plain old object. On handles retry automatically via thrown errors. If something 
+     * occurs in your callback and you'd like to try again, simply throw an uncaught error. The event will be retried
+     * after a configurable amount of time. If the event fails too many times, it can be sent to a dead letter queue
+     * for additional analysis. 
+     * 
+     * Check out the {@tutorial on-tutorial}
+     * 
+     * ### Required Environment Variables
+     * - REGION - This is the AWS region our service is operating in
+     * - ACCOUNT - The AWS account number we are using
+     * - DEPLOYMENT - The deployment we are part of
+     * - SERVICE - The name of our service
+     * 
+     * @module On
+     * @param {String} eventName The name of the event to listen for
+     * @param {Function} callback The function to call when the event is received
+     * @param {Object} config Optional configuration
+     * @param {String} config.queueGroup Allows listening to the same event from multiple spots
+     * 
+     * @return {Promise<boolean>} true if the listener is setup, false if the listener failed to listen
```

I wonder if it's better to throw an error rather than to return false if the listener fails to be setup. With a return value, we have to rely on the devs to wrap `on()` in an if or something. With an error they wouldn't be able to ignore failed listeners.